<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgMeshOptions</class>
 <widget class="QDialog" name="DlgMeshOptions">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>366</width>
    <height>420</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate mesh</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tabSurfaceSettings">
      <attribute name="title">
       <string>Component</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QComboBox" name="cbBody"/>
       </item>
       <item>
        <widget class="QCheckBox" name="cbUseDefaults">
         <property name="text">
          <string>Use default values</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbStretchedMesh">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Use hybrid mesh generation algorithm</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="textLabel1">
           <property name="text">
            <string>Maximum edge length</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="sbMaxLength">
           <property name="alignment">
            <set>Qt::AlignRight</set>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>99999999.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.200000000000000</double>
           </property>
           <property name="value">
            <double>0.800000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="textLabel1_2">
           <property name="text">
            <string>Minimum edge length</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="sbMinLength">
           <property name="alignment">
            <set>Qt::AlignRight</set>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>99999999.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.050000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="textLabel2">
           <property name="text">
            <string>Normal difference</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="sbMaxPhi">
           <property name="alignment">
            <set>Qt::AlignRight</set>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>60.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>2.000000000000000</double>
           </property>
           <property name="value">
            <double>30.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Triangle stretch ratio</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="sbMaxStretch">
           <property name="alignment">
            <set>Qt::AlignRight</set>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>2.500000000000000</double>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>6.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="lbLEFactor">
           <property name="text">
            <string>Leading edge refinement</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="sbLEFactor">
           <property name="alignment">
            <set>Qt::AlignRight</set>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>0.500000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
           <property name="value">
            <double>2.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="lbTEFactor">
           <property name="text">
            <string>Trailing edge refinement</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QDoubleSpinBox" name="sbTEFactor">
           <property name="alignment">
            <set>Qt::AlignRight</set>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>0.500000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
           <property name="value">
            <double>2.500000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabGlobalSettings">
      <attribute name="title">
       <string>Global</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Global settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QCheckBox" name="cbAllDefaults">
            <property name="text">
             <string>Use default values for all components</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Stretched triangle removal options</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Maximum stretch ratio</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="sbStretchFactor">
            <property name="minimumSize">
             <size>
              <width>90</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string comment="Collapse triangles exceeding this stretch ratio"/>
            </property>
            <property name="whatsThis">
             <string comment="Edges for which at least one of the neighbor triangles exceeds this strtch ratio will be collapsed into a point"/>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight</set>
            </property>
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>3.000000000000000</double>
            </property>
            <property name="maximum">
             <double>99.900000000000006</double>
            </property>
            <property name="singleStep">
             <double>0.500000000000000</double>
            </property>
            <property name="value">
             <double>7.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Normal angle tolerance</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="sbGlobalMaxPhi">
            <property name="minimumSize">
             <size>
              <width>90</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string comment="Keep triangles with larger difference in vertex normals"/>
            </property>
            <property name="whatsThis">
             <string comment="Keep stretched triangles in which the vertex normals differ by more than this angle"/>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight</set>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="maximum">
             <double>45.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>5.000000000000000</double>
            </property>
            <property name="value">
             <double>20.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Number of iterations</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="sbIterations">
            <property name="minimumSize">
             <size>
              <width>90</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string comment="Number of destretching passes to perform"/>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight</set>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>9</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout">
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>91</width>
         <height>27</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonCancel">
       <property name="text">
        <string>&amp;Cancel</string>
       </property>
       <property name="shortcut">
        <string/>
       </property>
       <property name="autoDefault">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonOk">
       <property name="text">
        <string>Create &amp;mesh</string>
       </property>
       <property name="shortcut">
        <string>Alt+M</string>
       </property>
       <property name="autoDefault">
        <bool>true</bool>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>cbBody</tabstop>
  <tabstop>cbUseDefaults</tabstop>
  <tabstop>cbStretchedMesh</tabstop>
  <tabstop>sbMaxLength</tabstop>
  <tabstop>sbMinLength</tabstop>
  <tabstop>sbMaxPhi</tabstop>
  <tabstop>sbLEFactor</tabstop>
  <tabstop>sbTEFactor</tabstop>
  <tabstop>cbAllDefaults</tabstop>
  <tabstop>sbStretchFactor</tabstop>
  <tabstop>sbGlobalMaxPhi</tabstop>
  <tabstop>sbIterations</tabstop>
  <tabstop>buttonCancel</tabstop>
  <tabstop>buttonOk</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonOk</sender>
   <signal>clicked()</signal>
   <receiver>DlgMeshOptions</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>394</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonCancel</sender>
   <signal>clicked()</signal>
   <receiver>DlgMeshOptions</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>394</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>textLabel1</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>25</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>textLabel1_2</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>214</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>139</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>textLabel2</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>222</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>150</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>lbLEFactor</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>180</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>lbTEFactor</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>164</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbMaxLength</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>194</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbMinLength</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>250</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbMaxPhi</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>256</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbLEFactor</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbTEFactor</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>label</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseDefaults</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbMaxStretch</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>279</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
